db/tikv/coprocessor.go

```diff
@@ -74,17 +73,76 @@ func (db *coprocessor) Read(ctx context.Context, table string, key string, field
 	if err != nil {
 		return nil, err
 	}
+
 	//return db.table.DecodeValue(result.GetData(), fields)
 	return make(map[string][]byte, len(fields)), nil
 }
 
 func (db *coprocessor) Scan(ctx context.Context, table string, startKey string, count int, fields []string) ([]map[string][]byte, error) {
+	// construct the req
+	req := kv.Request{}
+	req.Concurrency = 1
+	req.Tp = kv.ReqTypeDAG
+	req.KeyRanges = []kv.KeyRange{db.table.GetScanRange(startKey)}
+	dag := db.table.BuildDAGTableScanWithLimitReq(fields, count)
+	req.StartTs = dag.StartTs
+	data, err := dag.Marshal()
+	if err != nil {
+		return nil, err
+	}
+	req.Data = data
+	// send req
+	client := db.db.GetClient()
+	res := client.Send(ctx, &req, nil)
+	defer res.Close()
+	for {
+		resSubset, err := res.Next(ctx)
```

Comment:
resSubset is []byte or others? I prefer putting it in the result

Reply:
resSubset contains the []byte which is marshaled from SelectResponse.

Reply:
can we ensure we get the count items?

Reply:
if we must ensure the count for the scan, we should unmarshaled the resSubset.

Reply:
why? e.g. here you use Next to get N items, not equal to count? maybe we should use N to make the map.

Reply:
Each next returns the [copResultSubset](https://github.com/pingcap/tidb/blob/1a7e9ecf5201c29ebcf3c466e6864b3744e3450d/store/tikv/coprocessor.go#L500), we should unmarshal the []byte of copResultSubset to get the [SelectResponse] (https://github.com/pingcap/tipb/blob/b64a6ffa4ace41132bc10af8243086ea1024d46b/proto/select.proto#L28). Only from the SelectResponse, we can know how much items returned from this copResultSubset.